pkg/api/routes.go

```diff
@@ -1088,22 +1109,23 @@ func (rh *RouteHandler) DeleteBlob(response http.ResponseWriter, request *http.R
 
 	err = imgStore.DeleteBlob(name, digest)
 	if err != nil {
+		details := zerr.GetDetails(err)
 		if errors.Is(err, zerr.ErrBadBlobDigest) { //nolint:gocritic // errorslint conflicts with gocritic:IfElseChain
-			zcommon.WriteJSON(response,
-				http.StatusBadRequest,
-				apiErr.NewErrorList(apiErr.NewError(apiErr.DIGEST_INVALID, map[string]string{"digest": digest.String()})))
+			details["digest"] = digest.String()
+			e := apiErr.NewError(apiErr.DIGEST_INVALID).AddDetail(details)
+			zcommon.WriteJSON(response, http.StatusBadRequest, apiErr.NewErrorList(e))
 		} else if errors.Is(err, zerr.ErrRepoNotFound) {
-			zcommon.WriteJSON(response,
-				http.StatusNotFound,
-				apiErr.NewErrorList(apiErr.NewError(apiErr.NAME_UNKNOWN, map[string]string{"name": name})))
+			details["name"] = name
+			e := apiErr.NewError(apiErr.NAME_UNKNOWN).AddDetail(map[string]string{"name": name})
+			zcommon.WriteJSON(response, http.StatusNotFound, apiErr.NewErrorList(e))
 		} else if errors.Is(err, zerr.ErrBlobNotFound) {
-			zcommon.WriteJSON(response,
-				http.StatusNotFound,
-				apiErr.NewErrorList(apiErr.NewError(apiErr.BLOB_UNKNOWN, map[string]string{".String()": digest.String()})))
```

Comment:
Here and below I introduced the side effect of changing key from ".String()" -> digest.
Let me know if for some reason this is not needed.